src/restful/controllers/adminController.js

```diff
@@ -154,6 +155,64 @@ class AdminController {
       return util.send(res);
     }
   }
+
+  static async getAllOpportunities(req, res) {
+    const db = admin.firestore();
+  
+    try {
+      const applicationsSnapshot = await db.collectionGroup("applications").get();
+      const opportunityApplicationsCount = {};
+      
+      // Count the number of applications for each opportunity
+      applicationsSnapshot.forEach((doc) => {
+        const applicationData = doc.data();
+        const opportunityId = applicationData.opportunity_id;
+        if (opportunityApplicationsCount[opportunityId]) {
+          opportunityApplicationsCount[opportunityId]++;
+        } else {
+          opportunityApplicationsCount[opportunityId] = 1;
+        }
+      });
+  
+      // Fetch opportunities separately
+      const opportunitiesSnapshot = await db.collection("opportunities").where("status", "!=", "archived").get();
+  
+      // Fetch user data for each opportunity asynchronously
+      const opportunitiesDataPromises = opportunitiesSnapshot.docs.map(async (doc) => {
+        const opportunityData = doc.data();
+        const userId = opportunityData.user_id;
+        const userRef = db.collection("users").doc(userId);
+        const userDoc = await userRef.get();
+        const userData = userDoc.data();
+        const opportunityId = doc.id;
+  
+        return {
+          ...opportunityData,
+          numberOfApplications: opportunityApplicationsCount[opportunityId] || 0,
+          userName: userData ? userData.first_name + " " + userData.first_name : "Unknown",
```

Comment:
Let's change this to `full_name` as it might get mistaken for a user name. Also it should be `userData.first_name +" " + userData.last_name`